src/app/modules/shared_models/message_model.nim

```diff
@@ -186,11 +194,18 @@ QtObject:
     of ModelRole.MentionedUsersPks:
       result = newQVariant(item.mentionedUsersPks.join(" "))
 
+  proc updateItemAtIndex(self: Model, index: int) =
+    let ind = self.createIndex(index, 0, nil)
+    self.dataChanged(ind, ind, self.allKeys)
+
   proc findIndexForMessageId*(self: Model, messageId: string): int =
+    result = -1
+    if messageId.len == 0:
+      return
     for i in 0 ..< self.items.len:
       if(self.items[i].id == messageId):
-        return i
-    return -1
+        result = i
+        return
```

Comment:
Shouldn't this method return an `int`? Sorry, still a nim newb but how does this even compile? :)

Reply:
Nim is weird. You can both do `return X` or `result = X` for a return value.
In this case, Patryk set the default return with `result = -1`